tezos/messages/src/p2p/encoding/operations_for_blocks.rs

```diff
@@ -114,307 +114,62 @@ impl From<OperationsForBlocksMessage> for Vec<Operation> {
-                let mut right = None;
-                while let Some(key) = map.next_key()? {
-                    match key {
-                        Field::Path => {
-                            if path.is_some() {
-                                return Err(de::Error::duplicate_field("path"));
-                            }
-                            path = Some(map.next_value()?);
-                        }
-                        Field::Right => {
-                            if right.is_some() {
-                                return Err(de::Error::duplicate_field("right"));
-                            }
-                            right = Some(map.next_value()?);
-                        }
-                    }
-                }
-                let path = path.ok_or_else(|| serde::de::Error::missing_field("path"))?;
-                let right = right.ok_or_else(|| serde::de::Error::missing_field("right"))?;
-                Ok(PathLeft::new(path, right, Default::default()))
-            }
-        }
+// -----------------------------------------------------------------------------------------------
+#[derive(Clone, Serialize, Deserialize, PartialEq, Debug)]
+pub enum PathItem {
+    Right(PathRight),
+    Left(PathLeft),
+}
 
-        const FIELDS: &'static [&'static str] = &["path", "right"];
-        deserializer.deserialize_struct("PathLeft", FIELDS, PathLeftVisitor)
+impl PathItem {
+    pub fn right(left: Hash) -> PathItem {
+        PathItem::Right(PathRight::new(left, Default::default()))
+    }
+    pub fn left(right: Hash) -> PathItem {
+        PathItem::Left(PathLeft::new(right, Default::default()))
     }
 }
 
 // -----------------------------------------------------------------------------------------------
-#[derive(Clone, Deserialize, PartialEq, Debug)]
-pub enum Path {
-    Left(Box<PathLeft>),
-    Right(Box<PathRight>),
-    Op,
-}
+#[derive(Clone, Serialize, Deserialize, PartialEq, Debug)]
+pub struct Path(pub Vec<PathItem>);
 
 impl Path {
-    fn depth(&self) -> Option<RecursiveDataSize> {
-        match self {
-            Path::Left(l) => l.depth,
-            Path::Right(r) => r.depth,
-            Path::Op => Some(0),
-        }
-    }
-}
-
-/// Manual serializization ensures that path depth does not exceed max value
-impl Serialize for Path {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-    {
-        match self.depth() {
-            Some(d) if d <= PATH_MAX_DEPTH => (),
-            _ => {
-                use serde::ser::Error;
-                return Err(Error::custom("Path is too deep"));
-            }
-        }
-        match self {
-            Path::Left(h) => serializer.serialize_newtype_variant("Path", 0, "Left", h),
-            Path::Right(h) => serializer.serialize_newtype_variant("Path", 1, "Right", h),
-            Path::Op => serializer.serialize_unit_variant("Path", 2, "Op"),
-        }
+    pub fn op() -> Self {
```

Comment:
I am confused about this, why is it returning an empty vector? what is that representing?

Reply:
A `Path::Op` is used to denote the only element in the tree, I guess.